Mergin/plugin.py

```diff
@@ -47,12 +52,55 @@ def initGui(self):
         #     self.iface.browserModel().reload()
 
     def unload(self):
+        self.iface.projectRead.disconnect(set_project_variables)
+
         QgsApplication.instance().dataItemProviderRegistry().removeProvider(self.data_item_provider)
         self.data_item_provider = None
         # this is crashing qgis on exit
         # self.iface.browserModel().reload()
 
 
+def set_project_variables():
+    """
+    Sets project related mergin variables. Supposed to be called when any QGIS project is (re)loaded.
+    If a loaded project is not a mergin project, it suppose to have custom variables without mergin variables by default.
+    If a loaded project is invalid - doesnt have metadata, mergin variables are removed.
+    """
+    settings = QSettings()
+    settings.beginGroup('Mergin/localProjects/')
+    for key in settings.allKeys():
+        # Expecting [<namespace>, <project_name>, <path>]
```

Comment:
I am a bit confused what a key really contains ... from the comment I would expect a list with three items, but it seems it is a string and I don't understand the `key_parts[-1] == 'path'` below ... could you please comment in the code what could be a valid input `key` ?